### ISSUE TITLE
missing mountpoints on some drives on linux

### ISSUE THREAD

**Issue by squirrelpiles** [resin-io-modules/drivelist#74]
same issue as #68 

here is what is returned for my root drive:

description:"Hitachi HDT72101"
device:"/dev/sda"
mountpoint:null
name:"/dev/sda"
size:"931.5G"
system:true

Mountpoint is null. This drive is mounted via UUID:
UUID=7af69416-051a-4149-a3e8-9b59becb56f7 /               ext4    errors=remount-ro 0       1

I thought this was fixed recently?  now it is back

I have 3.1.0 installed...

**Comment by jviotti** [comment]
Hi @squirrelpiles ,

That's weird. Maybe there is a corner case we didn't consider.

What's the output of `blkid -s UUID -o value /dev/sda*` and `grep "^/dev/disk/by-uuid/7af69416-051a-4149-a3e8-9b59becb56f7" /proc/mounts | cut -d ' ' -f 2 | tr '\n' ','`?

**Comment by squirrelpiles** [comment]
dave@lucas:~$ blkid -s UUID -o value /dev/sda*
7af69416-051a-4149-a3e8-9b59becb56f7
4139d758-2cdd-4cd8-8455-e101ccadc2ca  <-- this is my swap

dave@lucas:~$ grep "^/dev/disk/by-uuid/7af69416-051a-4149-a3e8-9b59becb56f7" /proc/mounts
/dev/disk/by-uuid/7af69416-051a-4149-a3e8-9b59becb56f7 / ext4 rw,relatime,errors=remount-ro,data=ordered 0 0
/dev/disk/by-uuid/7af69416-051a-4149-a3e8-9b59becb56f7 /var/lib/docker/aufs ext4 rw,relatime,errors=remount-ro,data=ordered 0 0


it appears to work.  but here is another piece of info:  
when I first entered blkid -s UUID -o value /dev/sda*, it returned nothing.  So I figure I need to sudo it.  Then it worked.  But then it also then worked without sudo.

Googling and reading man page shows that blkid has a cache file.  I am thinking that there are times when the cache is out of date.   The cache on ubuntu is at /etc/blkid.tab

Also, many places seem to indicate that blkid should be run as root, so I think that is probably the issue.

Do this:  

1. delete /etc/blkid.tab
2. as a non-root user, do:  blkid -s UUID -o value /dev/sda*
3. you will get no results
4. now run:  sudo blkid -s UUID -o value /dev/sda*
5. you get results.
6. as a user again: blkid -s UUID -o value /dev/sda*
7. magic - you now get results.

I played around and it also seems that things like USB sticks etc *will* show with a non-root blkid command

I tried to use lsblk -O MOUNTPOINT,UUID   which returns the right info when run as root.  If you run as user, you get mountpoints but not UUIDs.   Which is sort of indicating that the UUIDs are not visible to non-root users?

Couldn't you derive all the info from a df command?

**Comment by jviotti** [comment]
Great observations!

I think `df` could be a pretty good alternative, I'll give it a go when I have access to a Linux box. 

What happens if you pass `/dev/null` as a cache for `blkid`? From the man page: `balked -c /dev/null -s UUID -o value /dev/sda*`.

**Comment by jviotti** [comment]
Hm, I just tried the steps you mentioned above, and couldn't reproduce. `blkid` seems to return the correct UUIDs as root and as non-root, with any cache file. Maybe you can custom `sudoers` settings or something like that?

In any case, I'll try to recreate my work using `df`, so it doesn't bite any other user.

**Comment by jviotti** [comment]
Turns out I can reproduce by forcing the tool to create a cache file in a custom location, and trying to re-use it back.

**Comment by jviotti** [comment]
Using something like `df --output=source,target | grep "^/dev/sda*"` and joining the values of the second column should be good enough.

**Comment by ilyabreev** [comment]
For me it doesn't find drive when `/proc/mounts` contains something like:

`/dev/disk/by-label/DOROOT / ext4 rw,relatime,errors=remount-ro,data=ordered 0 0`

**Comment by jviotti** [comment]
Hi @ilyabreev ,

What happens when you run the following command: `df --output=source,target`? Does that lists your drive with the corresponding mount point?

**Comment by ilyabreev** [comment]
Yeah, it perfectly does. 
As a temporary workaround I [added](https://github.com/serverdzen/drivelist/commit/ada17d5c7b24ae8b74af0b84fb8baa1cd5f4f6ad) search by label.

**Comment by jviotti** [comment]
Awesome, thanks lot, I'll publish a fix soon, and probably ping you to test it out if you don't mind!

**Comment by jviotti** [comment]
@ilyabreev Let me know if https://github.com/resin-io-modules/drivelist/pull/80 fixes the issue for you! If so, I'll merge and publish a new version with it right afterwards :)

**Comment by jviotti** [comment]
I did a slight change to the PR to account for multiple mount points.

**Comment by ilyabreev** [comment]
Thank you for such a quick fix.
Now `linux.sh` returns:

```
device: /dev/vda
description:
size: 21476933632
mountpoint: /,
name: /dev/vda
protected: False
system: True
```

It seems ok but look at the mountpoint. It has extra comma at the end of string.

**Comment by ilyabreev** [comment]
I think it's because of `tr` behaviour of replacing `\n` to `,`. It'll be better to strip that trailin comma.

**Comment by jviotti** [comment]
Hi @ilyabreev ,

You should be able to ignore it. The actual parser that transforms `linux.sh` output to JSON takes care of stripping it out. We were previously doing in the script itself, but was way easier to do it in the JavaScript side. You can run `example.js` in the top-level directory to confirm this.

**Comment by ilyabreev** [comment]
Ah, that's my fault, sorry. Got it.

**Comment by jviotti** [comment]
@ilyabreev I've released v3.2.1 to NPM with the fix! Thanks a lot for reporting and testing!

**Comment by ilyabreev** [comment]
@jviotti Thanks a lot! Updated and it works like a charm!
